provider/test-programs/cloudfunctions-function/src/function.js

```diff
@@ -0,0 +1,9 @@
+const functions = require('@google-cloud/functions-framework');
+
+// Register an HTTP function with the Functions Framework
+functions.http('helloGET', (req, res) => {
+  // Your code here
+
+  // Send an HTTP response
+  res.send('OK');
+});
```

Comment:
bit of a bummer to have to check in the src and the zip... I get why though; we need the zip for the test, but we want the source for diffs. Many providers will need some bundling for lambda/functions; might be worth pulling something into providertest to do basic bundling so we don't have to check in the zips. (Or maybe we should have something built in to the engine itself; since users have the same problem.)

Reply:
Yeah. There must be some Pulumi way to automatically archive some inline sources. I've looked at examples/ and we're testing  HttpCallbackFunction which is nice to use but I think it's the overlay/derivative resource, it's not the first-level proper resource.

Reply:
Found it! Removed the zip.